ambry-router/src/main/java/com/github/ambry/router/TtlUpdateOperation.java

```diff
@@ -368,9 +368,9 @@ private boolean shouldReplicateBlobAndRetry() {
     // 3. and at least one replica returned successful status.
     // 4. and error code precedence is over AmbryUnavailable. If we have one success and one delete, shouldn't do retry.
     RouterErrorCode errorCode = ((RouterException) operationException.get()).getErrorCode();
-    return (routerConfig.routerRepairWithReplicateBlobEnabled && operationTracker.hasNotFound()
-        && operationTracker.getSuccessCount() > 0
-        && getPrecedenceLevel(errorCode) >= getPrecedenceLevel(RouterErrorCode.AmbryUnavailable));
+    return (routerConfig.routerRepairWithReplicateBlobOnTtlUpdateEnabled && operationTracker.hasNotFound()
+        && operationTracker.getSuccessCount() > 0 && getPrecedenceLevel(errorCode) >= getPrecedenceLevel(
```

Comment:
Same qs, why do we need on-demand replication for any error other than `AmbryUnavailable` ?

Reply:
For TtlUpdate, if some replicas returns determined errors like BlobDeleted, BlobExpired etc, we don't need do the on-demand replication. The error is determined already.